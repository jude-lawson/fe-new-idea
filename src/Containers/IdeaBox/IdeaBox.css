.ideabox-container {
  padding-top: 8rem;
}

@media screen and (max-width: 500px) {
  .ideabox-container {
    padding-top: 0rem;
  }
}