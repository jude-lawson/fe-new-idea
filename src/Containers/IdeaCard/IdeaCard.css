.idea--card-container {
  margin: 3rem 6rem;
  border: 3px solid var(--primary-red);
  padding: 2rem;
  background: #fff;
  position: relative;
  border-radius: 1rem;
  min-height: 12rem;
}

.card--rmv-btn {
  position: absolute;
  top: 2rem;
  right: 2rem;
  border: 2px solid var(--primary-red);
  padding: 4px 8px;
  border-radius: 10px;
  color: var(--primary-blue);
}

.card--rmv-btn:hover {
  background: var(--primary-red);
  color: var(--primary-white);
  border: 2px solid var(--primary-blue);
}

.card--comment-btn {
  position: absolute;
  bottom: 2rem;
  right: 2rem;
  border: 2px solid var(--primary-red);
  padding: 4px 8px;
  border-radius: 10px;
  color: var(--primary-blue);
}

.card--comment-btn:hover {
  background: var(--primary-blue);
  color: var(--primary-white);
  border: 2px solid var(--primary-red);
}