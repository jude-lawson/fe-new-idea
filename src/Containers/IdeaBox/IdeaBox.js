import React, { Component } from 'react';
import { IdeaCard } from '../IdeaCard/IdeaCard';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getIdeas } from '../../api-calls/api-calls';
import Header from '../Header/Header';

export class IdeaBox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      allIdeas: []
    };
  }

  async componentDidMount() {
    const allIdeas = await getIdeas();
    this.displayIdeas(allIdeas);
    this.setState({allIdeas});
  }

  displayIdeas = () => {
    if (this.state.allIdeas) {
      const displayedIdeas = this.state.allIdeas.map((idea, index) => {
        return <IdeaCard {...idea} key={index} />;
      });
      return displayedIdeas;
    }
  }

  render() {
    return (
      <section className="app-container">
        <main>
          {this.displayIdeas(this.state.allIdeas)}
        </main>
      </section>
    );
  }
}

IdeaBox.propTypes = {
  ideas: PropTypes.array
};

export const mapStateToProps = (state) => ({
  ideas: state.ideas
});

export default connect(mapStateToProps)(IdeaBox);