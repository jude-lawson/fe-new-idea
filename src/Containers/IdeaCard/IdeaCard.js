import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import './IdeaCard.css';

export const IdeaCard = (props) => {
  const {title, body} = props;
  return (
    <article className="idea--card-container">
      <button className="card--rmv-btn">REMOVE</button>
      <h1 className="idea--card-title">{title}</h1>
      <p className="idea--card-body">{body}</p>
      <button className="card--comment-btn">COMMENT</button>
    </article>
  );
};

IdeaCard.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string
};
