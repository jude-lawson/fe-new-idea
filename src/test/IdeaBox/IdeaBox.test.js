import { IdeaBox } from '../../Containers/IdeaBox/IdeaBox';
import { shallow } from 'enzyme';
import React from 'react';
import { mapStateToProps } from '../../Containers/IdeaCard/IdeaCard';

describe('IdeaBox', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<IdeaBox />);

    expect(wrapper).toMatchSnapshot();
  })
});

describe('mapStateToProps', () => {
  it('should return a props object with the correct keys', () => {
    const mockState = {
      ideas: [
        {title: 'Whatever', body: 'Big Body Whatever', id: 11}
      ]
    }
    const expected = {
      ideas: [
        {title: 'Whatever', body: 'Big Body Whatever', id: 11}
      ]
    }
    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  })
})