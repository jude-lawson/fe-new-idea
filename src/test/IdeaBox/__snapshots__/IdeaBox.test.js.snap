// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IdeaBox /> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaBox
    getAllIdeas={[Function]}
    ideas={
        Array [
            Object {
              "body": "world",
              "id": 1,
              "title": "hello",
            },
            Object {
              "body": "one",
              "id": 2,
              "title": "another",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Unknown
        ideas={Array []}
/>,
      "className": "ideabox-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "ideas": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown
          ideas={Array []}
/>,
        "className": "ideabox-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "ideas": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
